### DISCUSSION TITLE
revalidatePath for static params

### DISCUSSION THREAD

**Discussion by tdejager** [vercel/next.js#50331] [App Router]
Hi!

Is there an easy way to call `revalidatePath` for a page that uses `generateStaticParams`. I.e revalidate all pages that have params that are generated from static params?

As I understand it you need to use the resolved page url e.g `blog/page/1` instead of `blog/page/[idx]` to do the resolution. However, that means you have to regenerate the static params in the revalidation route. Or is there are smarter way that I'm missing?

**Comment by bogadrian** [comment]
I hope you found a way. 
as far as I understand in order to revalidate a path the cache for that path must be deleted and rewritten. 
a path is a specific HTML page response from the server. If you want a specific page cached to be revalidated you must tell the server which one it needs to delete and rewrite. So yes, you need to tell it yiu need the page living in ca he at page/blog/1 to be revalidated - deleted and rewritten. There is no other way the server can know which page to revalidate. At least you do not to revalidate all the pages under that path.
